vapi/tags/tag_association.go

```diff
@@ -0,0 +1,114 @@
+// Copyright 2017 VMware, Inc. All Rights Reserved.
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//    http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+package tags
+
+import (
+	"context"
+	"encoding/json"
+	"fmt"
+	"net/http"
+)
+
+const (
+	TagAssociationURL = "/com/vmware/cis/tagging/tag-association"
+)
+
+type AssociatedObject struct {
+	ID   string `json:"id"`
+	Type string `json:"type"`
+}
+
+type TagAssociationSpec struct {
+	ObjectID *AssociatedObject `json:"object_id,omitempty"`
+	TagID    *string           `json:"tag_id,omitempty"`
+}
+
+func (c *RestClient) getAssociatedObject(objID string, objType string) *AssociatedObject {
+	if objID == "" && objType == "" {
+		return nil
+	}
+	object := AssociatedObject{
+		ID:   objID,
+		Type: objType,
+	}
+	return &object
+}
+
+func (c *RestClient) getAssociationSpec(tagID *string, objID string, objType string) *TagAssociationSpec {
+	object := c.getAssociatedObject(objID, objType)
+	spec := TagAssociationSpec{
+		TagID:    tagID,
+		ObjectID: object,
+	}
+	return &spec
+}
+
+func (c *RestClient) AttachTagToObject(ctx context.Context, tagID string, objID string, objType string) error {
```

Comment:
For these methods, instead of `tagID` and `objID` as string inputs, can we use 1 input `types.ManagedObjectReference` instead?